libs/contrib/Data/Combinators/Applicative.idr

```diff
@@ -0,0 +1,27 @@
+||| Enable applicative-style syntax (Mostly f <$> g <*> h and idiom brackets)
+||| for function composition:
+module Data.Combinators.Applicative
+
+import Data.Combinators
+
+%default total
+%access public export
+
+||| Pure is just const, also known as the K combinator
+pure : a -> b -> a
```

Comment:
`pure : Applicative f => a -> f a` is already defined as a member of the `Applicative` typeclass in Prelude.